apps/chatbot/src/modules/models.py

```diff
@@ -1,79 +1,89 @@
-        reason = kwargs.get("reason")
-        if reason == "GUARDRAIL_INTERVENED":
-            logging.info(f"Guardrails: {kwargs}")
-
-
-def get_llm(
-        params: dict, 
-        guardrails: Mapping[str, Any] | None = {
-            "guardrailIdentifier": AWS_GUARDRAIL_ID, 
-            "guardrailVersion": AWS_GUARDRAIL_VERSION, 
-            "trace": True
-        }
-    ):
-
-    if guardrails is None:
-        guardrails = {
-            "guardrailIdentifier": None,
-            "guardrailVersion": None,
-            "trace": None
-        }
-
-    # Guardrails for Amazon Bedrock with trace
-    llm = LangChainLLM(
-        BedrockLLM(
-            model_id = params["models"]["model_id"],
-            model_kwargs={
-                "temperature": params["models"]["temperature"],
-                "topP": params["models"]["topP"]
-            },
-            guardrails=guardrails,
-            callbacks = [BedrockAsyncCallbackHandler()],
-            streaming = params["engine"]["streaming"],
-            client = BEDROCK_CLIENT
+
+
+class ModelEventHandler(BaseEventHandler):
+    @classmethod
+    def class_name(cls) -> str:
+        """Class name."""
+        return "ModelEventHandler"
+
+    def handle(self, event) -> None:
+        """Logic for handling event."""
+        if isinstance(event, (LLMCompletionEndEvent, LLMChatEndEvent,)):
+            logging.info(f"Bedrock - RequestID: {event.response.raw["ResponseMetadata"]["RequestId"]}")
+            logging.info(f"Bedrock - Stop Reason: {event.response.raw["stopReason"]}")
+            logging.info(f"Bedrock - Input Tokens: {event.response.raw["usage"]["inputTokens"]}")
+            logging.info(f"Bedrock - Output Tokens: {event.response.raw["usage"]["outputTokens"]}")
+            logging.info(f"Bedrock - Latency (ms): {event.response.raw["metrics"]["latencyMs"]}")
+
+
+def get_llm(params: dict, add_event_handler: bool = True):
+
+    if params["models"]["provider"] == "aws":
+        if add_event_handler:
+            root_dispatcher = get_dispatcher()
+            root_dispatcher.add_event_handler(ModelEventHandler())
+        
+        llm = BedrockConverse(
+            model=params["models"]["model_id"],
+            temperature=params["models"]["temperature"],
+            max_tokens=params["models"]["max_tokens"],
+            aws_access_key_id=AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+            region_name=AWS_DEFAULT_REGION
         )
-    )
+
+    else:
+        llm = Gemini(
+            model=params["models"]["model_id"],
+            temperature=params["models"]["temperature"],
+            max_tokens=params["models"]["max_tokens"],
+            api_key=GOOGLE_API_KEY,
+        )
+
+    logging.info(f"{params["models"]["model_id"]} LLM loaded successfully!")
 
     return llm
 
 
-def get_embed_model(params):
+def get_embed_model(params: dict, provider: str ="aws"):
```

Comment:
Is the provider parameter really needed if we get it from the params?
```suggestion
def get_embed_model(params: dict):
```